app/uk/gov/hmrc/nisp/views/includes/continueWorking.scala.html

```diff
@@ -0,0 +1,78 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*@
+
+@import uk.gov.hmrc.nisp.models.{SPSummaryModel, SPChartModel}
+@import uk.gov.hmrc.nisp.views.formatting.{NispMoney, Time}
+@import uk.gov.hmrc.play.views.formatting.Dates
+@import uk.gov.hmrc.nisp.controllers.routes
+@import uk.gov.hmrc.nisp.utils.Constants
+
+@(spSummary: SPSummaryModel, currentChart: SPChartModel, forecastChart: SPChartModel)
+
+@if(spSummary.numberOfQualifyingYears < Constants.minimumQualifyingYearsNSP)  {
+
+    <div class="highlighted-event">
+        <p> @Messages("nisp.main.basedOn") @Dates.formatDate(spSummary.statePensionAge.date.localDate) @Messages("nisp.main.whenYoullBe") @spSummary.statePensionAge.age, @Messages("nisp.main.estimate").toLowerCase() @Messages("nisp.is")</p>
+        <p><em>@NispMoney.pounds(spSummary.forecast.forecastAmount.week) @Messages("nisp.main.week")</em></p>
+        <p> @NispMoney.pounds(spSummary.forecast.forecastAmount.month) @Messages("nisp.main.month"),
+            @NispMoney.pounds(spSummary.forecast.forecastAmount.year) @Messages("nisp.main.year")
+        </p>
+    </div>
+    <p>@Messages("nisp.main.estimate")</p>
+    <ul class="list-bullet">
+        <li>@Messages("nisp.main.notAGuarantee")</li>
+        <li>@Messages("nisp.main.isBased",Dates.formatDate(spSummary.lastProcessedDate.localDate))</li>
+        <li>@Messages("nisp.mqp.howManyToContribute",Time.years(spSummary.yearsToContributeUntilPensionAge))</li>
+        <li>@Messages("nisp.main.inflation")</li>
+    </ul>
+    <p> @Messages("nisp.mqp.youCurrentlyHave", Time.years(spSummary.numberOfQualifyingYears),Constants.minimumQualifyingYearsNSP)</p>
+    <div class="panel-indent">
+        <p>@Html(Messages("nisp.mqp.overseas"))</p>
+    </div>
+    <h3 class="heading-medium"> @NispMoney.pounds(spSummary.forecast.forecastAmount.week) @Messages("nisp.main.mostYouCanGet")</h3>
+    <p>@Messages("nisp.mqp.cannotImprove")</p>
+    <p>@Messages("nisp.main.after", Dates.formatDate(spSummary.statePensionAge.date.localDate))</p>
+
+} else {
+
+    <p>@Messages("nisp.main.estimate")</p>
+    <ul class="list-bullet">
+        <li>@Messages("nisp.main.notAGuarantee")</li>
+        <li>@Messages("nisp.main.inflation")</li>
+    </ul>
+
+    <h2 class="heading-medium">@Messages("nisp.main.h2.title")</h2>
+        @chart(Messages("nisp.main.chart.lastprocessed.title",spSummary.lastProcessedDate.localDate.getYear().toString()), currentChart.width, Html(s"${NispMoney.pounds(currentChart.amountModel.week)} ${Messages("nisp.main.chart.week")}"))
+
+        @if(spSummary.forecast.yearsLeftToWork == spSummary.yearsToContributeUntilPensionAge) {
```

Comment:
Instead of having ambiguous logic here we should use `Scenario.ContinueWorkingMax` and `Scenario.ContinueWorkingNonMax`, this is the part of the code they were designed for.